keel-web/src/main/kotlin/com/netflix/spinnaker/keel/services/ApplicationService.kt

```diff
@@ -0,0 +1,169 @@
+package com.netflix.spinnaker.keel.services
+
+import com.netflix.spinnaker.keel.api.DeliveryConfig
+import com.netflix.spinnaker.keel.core.api.ArtifactSummary
+import com.netflix.spinnaker.keel.core.api.ArtifactSummaryInEnvironment
+import com.netflix.spinnaker.keel.core.api.ArtifactVersionSummary
+import com.netflix.spinnaker.keel.core.api.ArtifactVersions
+import com.netflix.spinnaker.keel.core.api.EnvironmentSummary
+import com.netflix.spinnaker.keel.pause.ActuationPauser
+import com.netflix.spinnaker.keel.persistence.ArtifactRepository
+import com.netflix.spinnaker.keel.persistence.KeelRepository
+import com.netflix.spinnaker.keel.persistence.ResourceStatus
+import com.netflix.spinnaker.keel.persistence.ResourceSummary
+import com.netflix.spinnaker.kork.web.exceptions.InvalidRequestException
+import org.slf4j.LoggerFactory
+import org.springframework.stereotype.Component
+
+/**
+ * Service object that offers high-level APIs for application-related operations.
+ */
+@Component
+class ApplicationService(
+  private val pauser: ActuationPauser,
+  private val repository: KeelRepository,
+  private val artifactRepository: ArtifactRepository
+) {
+  private val log by lazy { LoggerFactory.getLogger(javaClass) }
+
+  fun hasManagedResources(application: String) = repository.hasManagedResources(application)
+
+  /**
+   * Returns a list of [ResourceSummary] for the specified application.
+   *
+   * This function assumes there's a single delivery config associated with the application.
+   */
+
+  fun getResourceSummariesFor(application: String): List<ResourceSummary> {
+    var resources = repository.getSummaryByApplication(application)
+    resources = resources.map { summary ->
+      if (pauser.resourceIsPaused(summary.id)) {
+        // we only update the status if the individual resource is paused,
+        // because the application pause is reflected in the response as a top level key.
+        summary.copy(status = ResourceStatus.PAUSED)
+      } else {
+        summary
+      }
+    }
+    // val constraintStates = repository.constraintStateFor(application)
+
+    return resources
+  }
+
+  fun getEnvironmentSummariesFor(application: String): List<EnvironmentSummary> =
+    getFirstDeliveryConfigFor(application)
```

Comment:
is the "first" delivery config deterministic? seems like we're making some assumptions here and i want to make sure those hold

Reply:
Not really, but what criteria do we have that I could use to pick the "right" one? Should we be failing these calls if there's more than one?

Reply:
P.S. https://github.com/spinnaker/keel/issues/676